trace-utils/src/send_data/mod.rs

```diff
@@ -60,243 +60,65 @@ pub enum RequestResult {
+///     retry_strategy::{RetryBackoffType, RetryStrategy},
+///     SendData,
+/// };
+/// use datadog_trace_utils::trace_utils::TracerHeaderTags;
+/// use ddcommon::Endpoint;
 /// use std::time::Duration;
 ///
-/// let retry_strategy = RetryStrategy {
-///     max_retries: 5,
-///     delay_ms: Duration::from_millis(100),
-///     backoff_type: RetryBackoffType::Exponential,
-///     jitter: Some(Duration::from_millis(50)),
-/// };
+/// #[cfg_attr(miri, ignore)]
+/// async fn update_send_results_example() {
+///     let size = 100;
+///     let tracer_payload = TracerPayload::default(); // Replace with actual payload
+///     let tracer_header_tags = TracerHeaderTags::default(); // Replace with actual header tags
+///     let target = Endpoint::default(); // Replace with actual endpoint
+///
+///     let mut send_data = SendData::new(size, tracer_payload, tracer_header_tags, &target);
+///
+///     // Set a custom retry strategy
+///     let retry_strategy = RetryStrategy::new(3, 10, RetryBackoffType::Exponential, Some(5));
+///
+///     send_data.set_retry_strategy(retry_strategy);
+///
+///     // Send the data
+///     let result = send_data.send().await;
+/// }
 /// ```
-#[derive(Debug, Clone)]
-pub struct RetryStrategy {
-    /// The maximum number of retries to attempt.
-    pub max_retries: u32,
-    /// The minimum delay between retries.
-    pub delay_ms: Duration,
-    /// The type of backoff to use for the delay between retries.
-    pub backoff_type: RetryBackoffType,
-    /// An optional jitter to add randomness to the delay.
-    pub jitter: Option<Duration>,
-}
-
-impl Default for RetryStrategy {
-    fn default() -> Self {
-        RetryStrategy {
-            max_retries: 5,
-            delay_ms: Duration::from_millis(100),
-            backoff_type: RetryBackoffType::Exponential,
-            jitter: None,
-        }
-    }
-}
-
-impl RetryStrategy {
-    /// Delays the next request attempt based on the retry strategy.
-    ///
-    /// If a jitter duration is specified in the retry strategy, a random duration up to the jitter
-    /// value is added to the delay.
-    ///
-    /// # Arguments
-    ///
-    /// * `attempt`: The number of the current attempt (1-indexed).
-    pub(crate) async fn delay(&self, attempt: u32) {
-        let delay = match self.backoff_type {
-            RetryBackoffType::Exponential => self.delay_ms * 2u32.pow(attempt - 1),
-            RetryBackoffType::Constant => self.delay_ms,
-            RetryBackoffType::Linear => self.delay_ms + (self.delay_ms * (attempt - 1)),
-        };
-
-        if let Some(jitter) = self.jitter {
-            let jitter = rand::random::<u64>() % jitter.as_millis() as u64;
-            sleep(delay + Duration::from_millis(jitter)).await;
-        } else {
-            sleep(delay).await;
-        }
-    }
-}
-
-#[derive(Debug, Clone)]
 pub struct SendData {
-    pub(crate) tracer_payloads: Vec<pb::TracerPayload>,
+    pub(crate) tracer_payloads: Vec<TracerPayload>,
```

Comment:
A bit late to the game here, but I think that the `pb::*` structs should always be fully qualified since some of the names are _generic_ and will be used by other types in these files. I mean there will likely be a `Span` that is not the `pb::Span` to represent some of these things. Anyway, nothing to change now, just something to keep in mind going forward.

Reply:
Ah, that's a good point. I slightly prefer aliasing the import in that situation, but this is fine too. Added back in #495